packages/RawButton/src/RawButton.js

```diff
@@ -62,48 +67,45 @@ const RawButton = React.forwardRef((props, ref) => {
   };
 
   function handleBlur() {
-    if ("onBlur" in moreProps) {
-      moreProps.onBlur();
-    }
+    if (moreProps.onBlur) moreProps.onBlur();
     setHasForcedFocus(false);
   }
 
   const handleKeyDown = event => {
-    if (
-      // Prevent scrolling the page with a spacerbar keypress
-      event.key === " " ||
-      // Prevent submitting forms in IE/Edge with and enter keypress
-      event.key === "Enter"
-    ) {
-      event.preventDefault();
-    }
+    // Prevent scrolling the page with a spacerbar keypress
+    // Prevent submitting forms in IE/Edge with and enter keypress
+    if (isTriggerKey(event.key)) event.preventDefault();
+    const shouldHandle = canPropagate || event.target === rawButtonRef.current;
+    if (shouldHandle && moreProps.onKeyDown) moreProps.onKeyDown(event);
   };
 
   const handleKeyUp = event => {
     const shouldHandle = canPropagate || event.target === rawButtonRef.current;
-    const isTriggerKey = event.key === " " || event.key === "Enter";
-
-    if (!isDisabled && shouldHandle && isTriggerKey) {
+    if (!isDisabled && shouldHandle && isTriggerKey(event.key)) {
       onClick(event);
     }
+    if (shouldHandle && moreProps.onKeyUp) moreProps.onKeyUp(event);
   };
 
   const bestTabIndex = isDisabled && tabIndex === null ? -1 : tabIndex || 0;
 
   return (
     <span
+      aria-pressed={isActive}
+      data-pka-anchor="raw-button"
+      tabIndex={bestTabIndex}
+      aria-label={a11yText}
+      {...moreProps}
```

Comment:
Just a thought, I wish there was a good way to know where the spreading of `moreProps` should be in the order of applied attributes here.... would be very easy for someone to move this around and not know if it breaks things.... 🤔

Reply:
Anything that would interfere with the proper workings of the component if it were overridden should be after `moreProps`.  Or if it just doesn't make sense to be overridden (ex `isDisabled={isDisabled}` – this _can't_ be overridden since `isDisabled` is already excluded from `moreProps`).  But properties that won't break the component and may have a valid use case to be overridden (ex `data-pka-anchor` may be overridden in other components that want to identify this component more specifically, like `dropdown-menu.trigger-button`).

Reply:
Short version: offer as much flexibility as possible, but restrict the ability to break it.